man/environment.d.xml

```diff
@@ -0,0 +1,107 @@
+<!--
+  This file is part of systemd.
+
+  Copyright 2016 Red Hat, Inc.
+
+  systemd is free software; you can redistribute it and/or modify it
+  under the terms of the GNU Lesser General Public License as published by
+  the Free Software Foundation; either version 2.1 of the License, or
+  (at your option) any later version.
+
+  systemd is distributed in the hope that it will be useful, but
+  WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public License
+  along with systemd; If not, see <http://www.gnu.org/licenses/>.
+-->
+<refentry id="environment.d" xmlns:xi="http://www.w3.org/2001/XInclude">
+
+  <refentryinfo>
+    <title>environment.d</title>
+    <productname>systemd</productname>
+
+    <authorgroup>
+      <author>
+        <contrib>Developer</contrib>
+        <firstname>Ray</firstname>
+        <surname>Strode</surname>
+        <email>rstrode@redhat.com</email>
+      </author>
+    </authorgroup>
+  </refentryinfo>
+
+  <refmeta>
+    <refentrytitle>environment.d</refentrytitle>
+    <manvolnum>5</manvolnum>
+  </refmeta>
+
+  <refnamediv>
+    <refname>environment.d</refname>
+    <refpurpose>Definition of user session environment</refpurpose>
+  </refnamediv>
+
+  <refsynopsisdiv>
+    <para><filename>~/.config/environment.d/*.conf</filename></para>
+    <para><filename>/run/user/<replaceable>uid</replaceable>/systemd/environment.d</filename></para>
+    <para><filename>/etc/environment.d/*.conf</filename></para>
+    <para><filename>/run/environment.d/*.conf</filename></para>
+    <para><filename>/usr/lib/environment.d/*.conf</filename></para>
+  </refsynopsisdiv>
+
+  <refsect1>
+    <title>Description</title>
+
+    <para>The <filename>environment.d</filename> directories contain a list of
+    "global" environment variable assignments for the user environment. Those files
+    are used by the systemd user instance, and should also be read by any
+    program which opens a user session
+    (<citerefentry project='man-pages'><refentrytitle>login</refentrytitle><manvolnum>1</manvolnum></citerefentry>,
+    <citerefentry project='die-net'><refentrytitle>ssh</refentrytitle><manvolnum>1</manvolnum></citerefentry>, etc.)
+    </para>
+
+    <para><command>systemd --user</command> will load those environment variables
+    when it is started or reloaded. They will be exported to all services.</para>
+  </refsect1>
+
+  <refsect1>
+    <title>Configuration Format</title>
+
+    <para>The configuration files contain a list of
+    <literal>KEY=VALUE</literal> environment variable assignments, separated
+    by newlines. The right hand side of these assignments may reference
+    previously defined environment variables, using the
+    <literal>${OTHER_KEY}</literal> format.</para>
```

Comment:
what precisely is the usecase for variable expansion here?

This has been requested before for EnvironmentFile=, but I so far have been very conservative about it. If we do add this here, why not also in EnvironmentFile=?

Is this syntax shell compatible? environment files so far always have been considered something you could also source in...